packages/agents/spade_agents/sm_agent.py

```diff
@@ -3,33 +3,91 @@
 import random
 import socket
 import time
+import math
 from asyncio.queues import Queue
+from copy import copy
 from dataclasses import asdict, dataclass
 
+import numpy as np
 from spade.agent import Agent
 from spade.behaviour import CyclicBehaviour, PeriodicBehaviour
 from spade.message import Message
 
+from .ws import WSServer
+
+
+def clamp(val, min_v, max_v):
+    return min(max(min_v, val), max_v)
+
 
 @dataclass
 class UserState:
+    x: float
+    y: float
     hp: float
-    mana: float
-    at: float
+    angle: float
+    velocity: float
+
+    @staticmethod
+    def default():
+        return UserState(0, 0, 0, 0, 0)
+
+
+class UserStateEncoder(json.JSONEncoder):
+    def default(self, z):
+        return asdict(z) if isinstance(z, UserState) else super().default(z)
+
+
+SIM_STEP = 0.1
 
 
 class SmartWatchAgent(Agent):
     class StateCollector(PeriodicBehaviour):
-        async def run(self) -> None:
-            state = UserState(hp=random.random(), mana=random.random(), at=random.random())
-            print(f"[[StateCollector]]: Fetching current user state: {state}")
-
-            if self.last_state != state:
-                await self.agent.state_broadcaster.my_user_state.put(state)
-                self.last_state = state
+        def __init__(self, *args, **kwargs) -> None:
+            super().__init__(*args, **kwargs)
+            self.last_state = UserState.default()
+            self.state = UserState(
+                x=random.random(),
+                y=random.random(),
+                hp=0.5 + random.random() / 2 * random.choice([-1, 1]),
+                angle=random.random() * math.pi * 2 - math.pi,
+                velocity=random.random(),
+            )
 
-        async def on_start(self) -> None:
-            self.last_state = UserState(0, 0, 0)
+        async def run(self) -> None:
+            radians = random.random() * math.pi * 2 - math.pi
+            velocity = (
+                1
+                - min(
+                    (2 * math.pi) - abs(radians - self.state.angle),
+                    abs(radians - self.state.angle),
+                )
+                / math.pi
+            )
+            self.state.x += math.sin(radians * math.pi) * velocity * SIM_STEP
+            self.state.y += math.cos(radians * math.pi) * velocity * SIM_STEP
+            self.state.angle = radians
+            self.state.velocity = velocity
+
+            self.state.x = clamp(self.state.x, 0, 1)
+            self.state.y = clamp(self.state.y, 0, 1)
+
+            rand_hp_factor = random.random()
+            if rand_hp_factor < 0.1:
+                hp_change_scale = 3.0
+            else:
+                hp_change_scale = 1.0
```

Comment:
Slightly more pythonic version of a same behaviour 🤗
```suggestion
            hp_change_scale = 3.0 if rand_hp_factor < 0.1 else 1.0
```